src/wal_replay.rs

```diff
@@ -215,6 +219,12 @@ impl WalReplayIterator<'_> {
             if let Some(sst_iter) = &mut self.current_iter.current_iter {
                 let wal_id = sst_iter.table_id().unwrap_wal_id();
                 while let Some(row_entry) = sst_iter.next_entry().await? {
+                    // skip the entries that are already in the L0 SST.
+                    if row_entry.seq <= self.min_seq {
```

Comment:
In the PR comments you said, "this pr is expected to not change any of the current behaviour of replaying/recovery," but it seems like this change does change the current behavior, right?

Reply:
the current code aligns exactly on the boundary of `wal sst` and `l0 sst`, when we replay the wal on wal, the seq of its entries should always be greater than the last l0's `last_wal_id`, so this filter should not affect the current behaviour i think 😲.

this code change relaxes that assumption, allowing for less strict alignment than previously required.